res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -1444,69 +1444,132 @@ let wheelTimerDelta = 0;
  * Kontrol S4 Mk3 hardware specific mapping logic
  */
 
+// A container for 2 effect units.
+// Uses Mixxx' standard effects mapping, supports effect focus mode.
+// https://manual.mixxx.org/2.5/en/chapters/effects#controller-effects-mapping
+// TODO implement auto-show/hide parameters option
 class S4Mk3EffectUnit extends ComponentContainer {
-    constructor(unitNumber, inReports, outReport, io) {
+    constructor(numbers, inReports, outReport, io) {
         super();
-        this.group = `[EffectRack1_EffectUnit${unitNumber}]`;
-        this.unitNumber = unitNumber;
+        if (!Array.isArray(numbers)) {
+            console.warn("ERROR! new S4Mk3EffectUnit() called without specifying any unit numbers!");
+            return;
+        }
+        if (numbers.length > 2) {
+            console.warn("ERROR! new S4Mk3EffectUnit() called with more than 2 unit numbers!");
+            return;
+        }
+        let numCount = 0;
+        for (const num of numbers) {
+            if (typeof num === "number") {
+                numCount++;
+            }
+        }
+        if (numCount < numbers.length) {
+            console.warn("ERROR! new S4Mk3EffectUnit() called with array that contains other types than numbers!");
+            return;
+        }
+        this.unitNumbers = numbers;
+
+        this.unitNumber = this.unitNumbers[0];
+        this.group = this.unitGroupForNumber(this.unitNumber);
         this.focusedEffect = null;
+        this.focusSelectMode = false;
+        // With this control skins can mark the unit that is currently mapped to the controller
+        engine.setValue(this.group, "controller_input_active", 1);
+        // Unset for the other unit
+        const otherUnitGroup = this.unitGroupForNumber(this.unitNumbers[1]);
+        engine.setValue(otherUnitGroup, "controller_input_active", 0);
```

Comment:
```suggestion
        // With this control skins can mark the unit that is currently mapped to the controller
        script.triggerControl(this.group, "controller_input_active");
        // Unset for the other unit
        const otherUnitGroup = this.unitGroupForNumber(this.unitNumbers[1]);
```

Reply:
umm, why?
we want it to be 1 for the 1st unit and 0 for the 2nd when the script is loaded.
`trigger` would flip it back to 0 after 200 ms.